src/main/java/eu/donals/UserInterface.java

```diff
@@ -0,0 +1,261 @@
+package eu.donals;
+
```

Comment:
This is a very complicated file compared to your other .java files. You could split it to improve maintainability. The comments do make it clear what everything does.